scripts/scraper-ng/rules/html-require-recipe-ingredients.js

```diff
@@ -0,0 +1,212 @@
+const fs = require("fs");
+const path = require("path");
+
+const { select } = require("hast-util-select");
+const visit = require("unist-util-visit");
+const yaml = require("js-yaml");
+
+/**
+ * A unified plugin that issues an error on pages that are missing ingredients.
+ */
+function requireRecipeIngredientsPlugin() {
+  return function warnOnMissingRecipeIngredients(tree, file) {
+    const recipe = loadRecipe(file.data.recipePath);
+
+    const requiredBody = recipe.body.filter(
+      ingredientName =>
+        !(ingredientName.endsWith("?") || ingredientName.endsWith(".*"))
+    );
+
+    let ok = true;
+    for (const ingredient of requiredBody) {
+      if (ingredient in ingredientHandlers) {
+        ok = ingredientHandlers[ingredient](tree, file) && ok;
+      } else {
+        file.message(`No handler for ingredient ${ingredient}`);
```

Comment:
This has no `message.ruleId` so it displays `null`. 
Should we add _something_ or are we fine with `null` in the report?

Reply:
👍 I'm fixing this specifically, but this brought to my attention more issues with `ruleId`s so I'm going to mention this in a much longer comment, forthcoming.